pkg/component/component.go

```diff
@@ -360,3 +362,91 @@ func Exists(client *occlient.Client, componentName string) (bool, error) {
 	}
 	return false, nil
 }
+
+// GetComponentService returns the Service object associated with the given
+// component.
+// An error is thrown when exactly one Service is not found for the component.
+func GetComponentService(client *occlient.Client, componentName string, applicationName string) (*corev1.Service, error) {
+	labels := GetLabels(componentName, applicationName, false)
+	selector := util.ConvertLabelsToSelector(labels)
+
+	services, err := client.GetServicesFromSelector(selector)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get Services for the selector: %v", selector)
+	}
+
+	numServices := len(services)
+	if numServices == 0 {
+		return nil, fmt.Errorf("no Service was found for the selector: %v", selector)
+	} else if numServices > 1 {
+		return nil, fmt.Errorf("multiple Services exist for the selector: %v. Only one must be present", selector)
+	}
+
+	return &services[0], nil
+}
+
+// LinkInfo contains the information about the link being created between
+// components
+type LinkInfo struct {
+	ComponentName string
```

Comment:
Is this source or target component name? 
Why is there only one?
There should be either comment, or better variable name should specify that.

Reply:
added source and target components